### DISCUSSION TITLE
How to have smooth playback experience in playlists?

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#4879] [FAQs] [answered]
The loading time between two mp4 files is long after creating the playlist with mp4 URLs, and the stream is not running smoothly. Please let me know if this can be resolved by changing some server settings.

**Comment by USAMAWIZARD** [comment, marked as answer]
Let me explain the best practices for that. I hope it helps.

*Improve WebRTC Playback Experience*

ATTENTION: It does not make sense to play the stream with WebRTC because it’s already recorded file and there is no ultra low latency requirement. It make sense to play the stream with HLS. Just keep in mind that WebRTC playback uses more processing resources than the HLS. Even if you would like to decrease the amount of time to switch streams, please read the followings.
1. Open the embedded player(`/usr/local/antmedia/webapps/{YOUR_APP}/play.html`) Find the `genericCallback` method and decrease the timeout value from `3000` to `1000` or even lower at the end of the `genericCallback` method. [<https://github.com/ant-media/StreamApp/blob/master/src/main/webapp/play.html#L261|It’s exactly this line>](https://github.com/ant-media/StreamApp/blob/c86d661f197238383b5eb75a421e58ae8fb6571e/src/main/webapp/play.html#L279)
2. Decrease the key frame interval of the video. You can set to 1 seconds. Generally recommend value is 2 seconds. WebRTC needs key frame to start the play. If the key frame interval is 10 seconds(default in ffmpeg), player may wait up to 10 seconds to play.

*Improve HLS Playback Experience*
1. Open the properties file of the application -&gt; `/usr/local/antmedia/webapps/{YOUR_APP}/WEB-INF/red5-web.properties`
2. Add the following property
settings.hlsflags=delete_segments+append_list+omit_endlist

Let me explain what it means.
`delete_segments` just deletes the segment files that is out of the list so that your disk will not get full.
`append_list` just adds the new segment files to the older m3u8 file so that player thinks that it’s just playing the same stream.
`omit_endlist` disables writing the EXT-X-ENDLIST to the end of the file so player thinks that new segments are in their way and it wait for them. It does not run stopping the stream.

3. Disable deleting hls files on ended to not encounter any race condition. Continue editing the file `/usr/local/antmedia/webapps/{YOUR_APP}/WEB-INF/red5-web.properties` and replace the following line `settings.deleteHLSFilesOnEnded=true` with this one
settings.deleteHLSFilesOnEnded=false

4. Restart the Ant Media Server
sudo service antmedia restart

**Comment by Mohit-3196** [comment]
Hi All,
Let me give an update on this starting from Ant Media Sever v2.6.2, the application settings be it in Standalone Mode or Cluster mode can be changed directly from the Web Panel without having to edit the application configuration file red5-web.properties and restarting the server.

For more details check the application configuration document https://antmedia.io/docs/guides/configuration-and-testing/ams-application-configuration/

Thank you.
